IyiKuyoro/Ride-My-Way
# #158791843 Use proper status codes and error messages for responses

### What does this PR do?
Improve the descriptiveness of json error messages
Improve the json responses format on all endpoints
#### Description of Task to be completed?
The following endpoints responses have been improved:
`/api/v1/auth/login`
`/api/v1/rides`
`/api/v1/rides/:rideId`
`/api/v1/rides/:rideId/requests`
`/api/v1/users/rides`
`/api/v1/users/rides/:rideId/requests`
`/api/v1/users/rides/:rideId/requests/:requestId`
#### How should this be manually tested?
Clone the repository first and start the server on a terminal with `npm run start`
  * Using postman, access the endpoints from localhost:3000
  * Ensure to pass the sign in token via the headers using a key 'jwt'
#### Any background context you want to provide?
These improvements where implemented based on feedback from my LFA
#### What are the relevant pivotal tracker stories?
  #158791843
  #158791784
